api/src/test/java/org/pmiops/workbench/api/DataSetControllerTest.java

```diff
@@ -442,13 +444,19 @@ public void setUp() throws Exception {
             QueryJobConfiguration.newBuilder(
                     "SELECT * FROM person_id from `${projectId}.${dataSetId}.person` person")
                 .build());
-    when(bigQueryService.filterBigQueryConfig(any()))
-        .thenReturn(
-            QueryJobConfiguration.newBuilder(
-                    "SELECT * FROM person_id from `" + TEST_CDR_TABLE + "` person")
-                .addNamedParameter(
-                    NAMED_PARAMETER_NAME, QueryParameterValue.string(NAMED_PARAMETER_VALUE))
-                .build());
+    // This is not great, but due to the interaction of mocks and bigquery, it is
+    // exceptionally hard to fix it so that it calls the real filterBitQueryConfig
+    // but _does not_ call the real methods in the rest of the bigQueryService
```

Comment:
Is it not possible to do

when(bigQueryService.filterBigQueryConfig(any())).thenCallRealMethod()

?

Reply:
Talked offline. Tried doing that. I suspect it has to do with `MockBean` and the way that spring and mockito interact in our tests.